### DISCUSSION TITLE
Question: How to keep multiple instances of a Page type in Frame?

### DISCUSSION THREAD

**Discussion by HppZ** [microsoft/microsoft-ui-xaml#7436] [Q&A]
I want every instance of a specific Page type to keep their own UI and data state in back stack of a Frame.

**Comment by seven-mile** [comment]
Having tested this situation on my own app, I believe the page instance in the navigation stack won't lost its state after navigating to a new page with the same type but different NavigationParameter. I don't know whether the parameter matters. You can test it, for example, by generate a Guid parameter for each page navigation.

By the way, if you are trying using this feature for something except just Navigate Back, probably you are using the wrong control.

**Comment by StephenLPeters** [comment]
@HppZ have you seen https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.page.navigationcachemode?view=winrt-22000?

**Comment by HppZ** [comment]
@seven-mile

https://github.com/HppZ/TestFrame

If I comment out this line, it works, but in my real app, the parameter is used to request data, when I go back to a previous page, I have to reload data and scroll ListView to previous offset and do many other things to keep every thing unchanged.
How to make Frame not reuse MainPage instance? If I don't set NavigationCacheMode to Required, then Frame creates new MainPage instance every time when I go back, I just need go back to the exact previous MainPage instance, not a new one, how to?

![image](https://user-images.githubusercontent.com/12285582/139372591-6e9d2dd4-7f62-49b5-bae9-7e4526e55f77.png)

**Comment by HppZ** [comment]
> @HppZ have you seen https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.page.navigationcachemode?view=winrt-22000?

Yes.

**Comment by StephenLPeters** [comment]
@codendone do you know?

**Comment by HppZ** [comment]
@chingucoding could you help me?

**Comment by marcelwgn** [comment]
@HppZ Not familiar with this, sorry
